pyrasterframes/src/main/python/docs/supervised-learning.pymd

```diff
@@ -0,0 +1,197 @@
+# Supervised Machine Learning
+
+In this example we will demonstrate how to fit and score an unsupervised learning model with a sample of Sentinel-2 data and labels from the US [National Land Cover Dataset](https://www.mrlc.gov/) (NLCD).
+
+```python, setup, echo=False
+from IPython.core.display import display
+from pyrasterframes.utils import create_rf_spark_session
+from pyrasterframes.rasterfunctions import * 
+import pandas as pd
+from docs import resource_dir_uri
+
+import os
+
+spark = create_rf_spark_session()
+```
+
+## Create and Read Raster Catalog
+
+We import various Spark components that we need to construct our [Pipeline](https://spark.apache.org/docs/latest/ml-pipeline.html).
+
+```python, imports, echo=True
+from pyrasterframes import TileExploder
+from pyrasterframes.rf_types import NoDataFilter
+
+from pyspark.ml.feature import VectorAssembler
+from pyspark.ml.classification import DecisionTreeClassifier
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+from pyspark.ml import Pipeline
+```
+
+The first step is to create a Spark DataFrame of our imagery data. To achieve that we will create @ref:[a catalog DataFrame](raster-catalogs.md#creating-a-catalog). In the catalog, each row represents a distinct area and time; and each column is the URI to a band's image product. The resulting Spark DataFrame may have many rows per URI, with a column corresponding to each band.
+
+```python, read_bands, term=True
+uri_base = 's3://s22s-test-geotiffs/luray_snp/{}.tif'
+bands = ['B01', 'B02', 'B03', 'B04', 'B05', 'B06', 'B07', 'B08', 'B09', 'B11', 'B12']
+cols = ['SCL'] + bands
+
+catalog_df = pd.DataFrame([
+    {b: uri_base.format(b) for b in cols} 
+])
+
+df = spark.read.raster(catalog=catalog_df, 
+					   catalog_col_names=cols,
+					   tile_dimensions=(128, 128)
+					   ).repartition(100)
+
+df = df.select(
+    rf_crs(df.B01).alias('crs'),
+    rf_extent(df.B01).alias('extent'),
+    rf_tile(df.SCL).alias('scl'),
+    rf_tile(df.B01).alias('B01'),
+    rf_tile(df.B02).alias('B02'),
+    rf_tile(df.B03).alias('B03'),
+    rf_tile(df.B04).alias('B04'),
+    rf_tile(df.B05).alias('B05'),
+    rf_tile(df.B06).alias('B06'),
+    rf_tile(df.B07).alias('B07'),
+    rf_tile(df.B08).alias('B08'),
+    rf_tile(df.B09).alias('B09'),
+    rf_tile(df.B11).alias('B11'),
+    rf_tile(df.B12).alias('B12'),
+)
+df.printSchema()
+```
+
+## Data Prep
+
+### Label Data
+
+[](https://github.com/locationtech/rasterframes/blob/develop/core/src/test/resources/L8-Labels-Elkton-VA.geojson)
```

Comment:
I've been trying to use stable GitHub "raw" URLs, but it ends up being a PITA to get the real ones (due to the use of redirects). We should think of a better way post release.

Reply:
here i was ok with the `blob` view because it does render nicely and it is merely for reader's convenience.